### PR TITLE
Aswathy/Topics clickable

### PR DESCRIPTION
**Pull request by aswathy-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/academy/search/index.tsx [hunk 1]
```diff
@@ -312,6 +312,98 @@ const SearchPage = () => {
...
+
+                {full_video_link && !items_type && total_video > 1 && (
+                    <AllArticleButton tertiary="true" to={full_video_link}>
+                        All video results
+                    </AllArticleButton>
+                )}
+            </>
+        )
+    }
+    const urlLocation = (url, is_search = false) => {
```
Comment (prince-deriv) [inline src/pages/academy/search/index.tsx:394]: can we change `urlLocation` name to `checkFilterType` or to anything more relevant

Comment (aswathy-deriv) [reply src/pages/academy/search/index.tsx]: okay changed

**Comment by mohammad-rework** [comment]
In `src/common/utility.js` **L377** we need to add isBrowser() function:
`        const params = new URLSearchParams(isBrowser() && location.search)
`

**Review by mohammad-rework — APPROVED** [review]

File: src/pages/academy/search/index.tsx [hunk 2]
```diff
@@ -312,6 +312,106 @@ const SearchPage = () => {
...
+        const is_article = checkFilterType('article')
+        const is_search = checkFilterType('', true)
+        const is_video = checkFilterType('video')
+        if (is_article || is_search) {
+            return true
+        }
+        if (is_video) {
+            return false
+        }
+    }
```
Comment (prince-deriv) [inline src/pages/academy/search/index.tsx:414]: We can remove the `checkFilterType` function and do the comparison directly in `isShowArticle` function

Comment (aswathy-deriv) [reply src/pages/academy/search/index.tsx]: okay

File: src/pages/academy/search/index.tsx [hunk 3]
```diff
@@ -312,6 +312,96 @@ const SearchPage = () => {
...
+    const isShowArticle = () => {
+        const type = queryParams.get('type')
+        if (type === 'article' || !type) {
+            return true
+        }
+
+        if (type === 'video') {
+            return false
+        }
+    }
```
Comment (prince-deriv) [inline src/pages/academy/search/index.tsx:404]: can be simplified in just one `return `

File: src/pages/academy/search/index.tsx [hunk 4]
```diff
@@ -312,6 +312,91 @@ const SearchPage = () => {
...
+                {full_video_link && !items_type && total_video > 1 && (
+                    <AllArticleButton tertiary="true" to={full_video_link}>
+                        All video results
+                    </AllArticleButton>
+                )}
+            </>
+        )
+    }
+
+    const isShowArticle = () => {
```
Comment (prince-deriv) [inline src/pages/academy/search/index.tsx:395]:
```suggestion
    const isShowArticleFirst = () => {
```

Comment (prince-deriv) [reply src/pages/academy/search/index.tsx:395]: `isShowArticle` might create confusion to other developers in the future.

**Review by prince-deriv — APPROVED** [review]

**Review by vijayasree-deriv — APPROVED** [review]

**Pull request merged**